src/read.jl

```diff
@@ -39,33 +41,27 @@ to `identity`.
 edf_unknown(f, field::AbstractString) = field == "X" ? missing : f(field)
 edf_unknown(field::AbstractString) = edf_unknown(identity, field)
 
-"""
-    set!(io::IO, [f=strip,] sigs::Vector{Signal}, name::Symbol, sz::Integer)
+#####
+##### Reading EDF Files
+#####
 
-For each signal in `sigs`, set the field given by `name` to the result of `f` applied
-to a chunk of size `sz` read from `io` as a `String`.
-"""
-function set!(io::IO, f, sigs::Vector{Signal}, name::Symbol, sz::Integer)
-    n = length(sigs)
-    T = fieldtype(Signal, name)
-    @inbounds for i = 1:n
-        x = f(String(Base.read(io, sz)))
-        setfield!(sigs[i], name, convert(T, x))
+
+function read_header(io::IO)
+    (file_header, header_byte_count, signal_count) = read_file_header(io)
```

Comment:
```suggestion
    file_header, header_byte_count, signal_count = read_file_header(io)
```

Reply:
yep, makes sense.

Reply:
Fixed in https://github.com/beacon-biosignals/EDF.jl/pull/23/commits/55d306674690b41d1b77f244476f31e1c9ab5b63 along with several other instances.